hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/CompactionService.java

```diff
@@ -0,0 +1,159 @@
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.service;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.hadoop.hdds.utils.BackgroundService;
+import org.apache.hadoop.hdds.utils.BackgroundTask;
+import org.apache.hadoop.hdds.utils.BackgroundTaskQueue;
+import org.apache.hadoop.hdds.utils.BackgroundTaskResult;
+import org.apache.hadoop.hdds.utils.db.RDBStore;
+import org.apache.hadoop.hdds.utils.db.RocksDatabase;
+import org.apache.hadoop.hdds.utils.db.managed.ManagedCompactRangeOptions;
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.OzoneManager;
+import org.apache.hadoop.util.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is the background service to compact OM rocksdb tables.
+ */
+public class CompactionService extends BackgroundService {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(CompactionService.class);
+
+  // Use only a single thread for Compaction.
+  private static final int COMPACTOR_THREAD_POOL_SIZE = 1;
+
+  private final OzoneManager ozoneManager;
+  private final OMMetadataManager omMetadataManager;
+  private final AtomicLong numCompactions;
+  private final AtomicBoolean suspended;
+  // list of tables that can be compacted
+  private final List<String> compactableTables;
+
+  public CompactionService(OzoneManager ozoneManager, TimeUnit unit, long interval, long timeout,
+                           List<String> tables) {
+    super("CompactionService", interval, unit,
+        COMPACTOR_THREAD_POOL_SIZE, timeout,
+        ozoneManager.getThreadNamePrefix());
+    this.ozoneManager = ozoneManager;
+    this.omMetadataManager = this.ozoneManager.getMetadataManager();
+
+    this.numCompactions = new AtomicLong(0);
+    this.suspended = new AtomicBoolean(false);
+    for (String table : tables) {
+      if (!omMetadataManager.listTableNames().contains(table)) {
+        throw new IllegalArgumentException();
+      }
+    }
+    this.compactableTables = tables;
+  }
```

Comment:
Can we not throw an exception if a table is invalid? Doing so would prevent the entire background compaction service from running. Instead, we should log the issue and proceed with compaction for the valid tables.
```suggestion
    this.compactableTables = validateTables(tables);
    LOG.info("CompactionService initialized with tables: {}", compactableTables);
  }

  private List<String> validateTables(List<String> inputTables) {
    List<String> validTables = new ArrayList<>();
  Set<String> allTableNames = new HashSet<>(omMetadataManager.listTableNames());
  for (String table : tables) {
    if (allTableNames.contains(table)) {
      validTables.add(table);
    } else {
      LOG.warn("CompactionService: Table \"{}\" not found in OM metadata. Skipping this table.", table);
    }
  }
  if (validTables.isEmpty()) {
    LOG.error("CompactionService: No valid compaction tables found. Failing initialization.");
    throw new IllegalArgumentException("CompactionService: None of the provided tables are valid.");
  }
  return Collections.unmodifiableList(validTables);
  }
```

Reply:
Thanks for the review @aryangupta1998 , i have incorporated your suggestion